Index: C/wcs.c
===================================================================
--- C/wcs.c
+++ C/wcs.c
@@ -1688,7 +1688,8 @@
     }

     /* Process linear axes. */
-    if (!(strlen(ctypei) == 8 && ctypei[4] == '-')) {
+    if (!(strlen(ctypei) == 8 && ctypei[4] == '-') ||
+        (strlen(ctypei) > 8 && ctypei[8] == '-')) {
       /* Identify Stokes, celestial and spectral types. */
       if (strcmp(ctypei, "STOKES") == 0) {
         /* STOKES axis. */
