Index: C/wcsfix.c
===================================================================
--- C/wcsfix.c
+++ C/wcsfix.c
@@ -61,7 +61,9 @@ const char *wcsfix_errmsg[] = {
   "Ill-conditioned coordinate transformation parameters",
   "All of the corner pixel coordinates are invalid",
   "Could not determine reference pixel coordinate",
-  "Could not determine reference pixel value"};
+  "Could not determine reference pixel value",
+  "Applied unit alias",
+};

 /* Convenience macro for invoking wcserr_set(). */
 #define WCSFIX_ERRMSG(status) WCSERR_SET(status), wcsfix_errmsg[status]
@@ -379,12 +381,23 @@ int unitfix(int ctrl, struct wcsprm *wcs)

 {
   int  i, status = FIXERR_NO_CHANGE;
+  char orig_unit[80];
+  const char *function = "unitfix";
+  struct wcserr **err;

   if (wcs == 0x0) return FIXERR_NULL_POINTER;
+  err = &(wcs->err);

   for (i = 0; i < wcs->naxis; i++) {
+    strncpy(orig_unit, wcs->cunit[i], 80);
     if (wcsutrne(ctrl, wcs->cunit[i], &(wcs->err)) == 0) {
-      status = FIXERR_SUCCESS;
+      if (strncmp(orig_unit, wcs->cunit[i], 80) != 0) {
+        status = wcserr_set(WCSERR_SET(FIXERR_APPLIED_UNIT_ALIAS),
+          "Applied unit alias '%s' -> '%s'",
+          orig_unit, wcs->cunit[i]);
+      } else {
+        status = FIXERR_SUCCESS;
+      }
     }
   }

Index: C/wcsfix.h
===================================================================
--- C/wcsfix.h
+++ C/wcsfix.h
@@ -366,8 +366,9 @@ enum wcsfix_errmsg_enum {
 				   invalid. */
   FIXERR_NO_REF_PIX_COORD =  9,	/* Could not determine reference pixel
 				   coordinate. */
-  FIXERR_NO_REF_PIX_VAL   = 10	/* Could not determine reference pixel
+  FIXERR_NO_REF_PIX_VAL   = 10,	/* Could not determine reference pixel
 				   value. */
+  FIXERR_APPLIED_UNIT_ALIAS = 11 /* Applied a unit alias. */
 };

 int wcsfix(int ctrl, const int naxis[], struct wcsprm *wcs, int stat[]);
