--- a/wcslib/C/wcs.c
+++ b/wcslib/C/wcs.c
@@ -2037,7 +2037,7 @@ int wcs_units(struct wcsprm *wcs)
       if (wcsunitse(wcs->cunit[i], units, &scale, &offset, &power,
                     &uniterr)) {
         wcserr_set(WCSERR_SET(WCSERR_BAD_COORD_TRANS),
-          "In CUNIT%d%.1s: %s", i, (*wcs->alt)?wcs->alt:"", uniterr->msg);
+          "In CUNIT%d%.1s: %s", i+1, (*wcs->alt)?wcs->alt:"", uniterr->msg);
         free(uniterr);
         return WCSERR_BAD_COORD_TRANS;
       }
