--- a/wcslib/C/wcsfix.c
+++ b/wcslib/C/wcsfix.c
@@ -417,6 +417,7 @@ int unitfix(int ctrl, struct wcsprm *wcs)
   char orig_unit[80], msg[WCSERR_MSG_LENGTH];
   const char *function = "unitfix";
   struct wcserr **err;
+  int result;

   if (wcs == 0x0) return FIXERR_NULL_POINTER;
   err = &(wcs->err);
@@ -424,7 +425,8 @@ int unitfix(int ctrl, struct wcsprm *wcs)
   strcpy(msg, "Changed units: ");
   for (i = 0; i < wcs->naxis; i++) {
     strncpy(orig_unit, wcs->cunit[i], 80);
-    if (wcsutrne(ctrl, wcs->cunit[i], &(wcs->err)) == 0) {
+    result = wcsutrne(ctrl, wcs->cunit[i], &(wcs->err));
+    if (result == 0 || result == 12) {
       k = strlen(msg);
       sprintf(msg+k, "'%s' -> '%s', ", orig_unit, wcs->cunit[i]);
       status = FIXERR_UNITS_ALIAS;
